src/Microsoft.AspNetCore.Routing/Internal/NullRouteConstraint.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Internal
```

Comment:
No need for `.Internal` here. This should live with the other route constraint implementations.